### ISSUE TITLE
`tfe_project` data source not finding projects

### ISSUE THREAD

**Issue by pcantea** [hashicorp/terraform-provider-tfe#882]
#### Terraform Cloud

The `tfe_project` data source does not find existing projects.

#### Terraform version

```plaintext
1.4.6
```

#### tfe provider version
```plaintext
0.44.1
```

#### Terraform Configuration Files

```terraform
data "tfe_project" "foo" {
  organization = "org"
  name         = "project"
}

```

#### Debug Output

```plaintext
data.tfe_project.foo: Refreshing...
╷
│ Error: could not find project org/project
│
│   with data.tfe_project.foo,
│   on main.tf line 1, in data "tfe_project" "foo":
│    1: data "tfe_project" "foo" {
│
╵
```

#### Expected Behavior

Expected project to be found

#### Actual Behavior

Project not found

**Comment by olliefr** [comment]
@pcantea I think you might need to set a token for the `tfe` provider if you'd like to run your Terraform configuration on Terraform Cloud [docs](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs#authentication)

**Comment by SylvainMartel** [comment]
I have the same behavior, it cannot see the project.  I use a user token that is "owner"(admin) in TF Cloud.  I did both a successful terraform login and also set the TFE_TOKEN variable just to be sure.

**Comment by felipefrocha** [comment]
For me as well, right now I am org owner and still not able to retrieve project data as well.
I just change to that because of other issue. When I imported the project to terraform state it was possible, but when I tried to apply the changes, it loses this resource states and try to create again, which results in error once the project already exists in that organization.

**Comment by JensBEvolate** [comment]
Had the same issue when I moved working code from local backend to terraform cloud.

When setting the token in the  provider "tfe" it worked.

```
provider "tfe" {
  token = var.token
}
```

**Comment by brandonc** [comment]
I was not able to reproduce the the finding that `tfe_project` cannot be used to read a project unless the token is configured on the provider. It must be the case that the token being configured on the provider has more permissions that another token configured through an environment variable. You can test this by using a TFE_TOKEN variable to create and then read a project:

```terraform
provider "tfe" {
  # configured TFE_TOKEN
}

resource "tfe_project" "foo" {
  name = "foo"
  organization = "example"
}

data "tfe_project" "project" {
  name = tfe_project.foo.name
  organization = "example"
}

output "project_id" {
  value = data.tfe_project.project.id
}
```

**Comment by archepro84** [comment]
I am experiencing the same issue.

I have configured `data "tfe_project"` like `resource "tfe_project"` that Terraform imported, but I am getting the issue of `could not find project <Organization Name>/<Project Name>`.

I tried to counteract the "Reissue API Token" and "Modify User Permissions" as suggested by others above, but it didn't work.

**Terraform Plan Error Message**
```
│ Error: could not find project <Organization Name>/<Project Name>
│ 
│ with data.tfe_project.this,
│ on outputs.tf line 1, in data "tfe_project" "this":
│ 1: data "tfe_project" "this" {
```
